dev/tools/controllerbuilder/pkg/llm/ollama.go

```diff
@@ -0,0 +1,296 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package llm
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"os"
+
+	"k8s.io/klog/v2"
+)
+
+// BuildOllamaClient builds a client for the Ollama API.
+func BuildOllamaClient(ctx context.Context) (*OllamaClient, error) {
```

Comment:
What different features ollama gives us compared to vertextAI?

Reply:
So ollama lets us easily use self-hosted models like gemma.  gemma doesn't currently support function calling (AFAICT) so we're actually somewhat limited in the models we can use here, but it lets us benchmark more models.  Also, as a library, we want to be able to support more models and ollama gets us a lot!